python-checks/src/main/java/org/sonar/python/checks/hotspots/CsrfDisabledCheck.java

```diff
@@ -0,0 +1,326 @@
+    "flask_wtf.csrf.CSRFProtect.exempt"));
+
+  /** Raises issue whenever a decorator with something about "CSRF" and "exempt" in the combined name is found. */
+  private static void decoratorCsrfExemptCheck(SubscriptionContext subscriptionContext) {
+    Decorator decorator = (Decorator) subscriptionContext.syntaxNode();
+    List<String> names = decorator.name().names().stream().map(Name::name).collect(Collectors.toList());
+    // This is a temporary workaround until symbol resolution works for decorators.
+    // Use the actual functions with FQNs from DANGEROUS_DECORATORS once that's fixed.
+    // Related ticket: https://jira.sonarsource.com/browse/SONARPY-681
+    boolean isDangerous = names.stream().anyMatch(s -> s.toLowerCase(Locale.US).contains("csrf")) &&
+      names.stream().anyMatch(s -> s.toLowerCase(Locale.US).contains("exempt"));
+    if (isDangerous) {
+      subscriptionContext.addIssue(decorator.lastToken(), DISABLING_CSRF_MESSAGE);
+    }
+  }
+
+  /** Raises an issue whenever one of the CSRF-exemption decorators is used as an ordinary function. */
+  private static void functionCsrfExemptCheck(SubscriptionContext subscriptionContext) {
+    CallExpression callExpr = (CallExpression) subscriptionContext.syntaxNode();
+    Optional.ofNullable(callExpr.calleeSymbol())
+      .map(Symbol::fullyQualifiedName)
+      .filter(DANGEROUS_DECORATORS::contains)
+      .ifPresent(fqn -> subscriptionContext.addIssue(callExpr.callee().lastToken(), DISABLING_CSRF_MESSAGE));
+  }
+
+  /** Checks that <code>'WTF_CSRF_ENABLED'</code> setting is not switched off. */
+  private static void flaskWtfCsrfEnabledFalseCheck(SubscriptionContext subscriptionContext) {
+    AssignmentStatement asgn = (AssignmentStatement) subscriptionContext.syntaxNode();
+    // Checks that the left hand side is some kind of subscription of `something['WTF_CSRF_ENABLED']`
+    // Does not check what `something` is - overtainting seems extremely unlikely in this case.
+    boolean isWtfCsrfEnabledSubscription = asgn
+      .lhsExpressions()
+      .stream()
+      .flatMap(exprList -> exprList.expressions().stream())
+      .filter(expr -> expr.is(Tree.Kind.SUBSCRIPTION))
+      .flatMap(s -> ((SubscriptionExpression) s).subscripts().expressions().stream())
+      .anyMatch(isStringSatisfying(s -> "WTF_CSRF_ENABLED".equals(s) || "WTF_CSRF_CHECK_DEFAULT".equals(s)));
+    if (isWtfCsrfEnabledSubscription && Expressions.isFalsy(asgn.assignedValue())) {
+      subscriptionContext.addIssue(asgn.assignedValue(), DISABLING_CSRF_MESSAGE);
+    }
+  }
+
+  /**
+   * Detects <code>class Meta</code> withing <code>FlaskForm</code>-subclasses,
+   * with <code>csrf</code> set to <code>False</code>.
+   */
+  private static void metaCheck(SubscriptionContext subscriptionContext) {
+    ClassDef classDef = (ClassDef) subscriptionContext.syntaxNode();
+    boolean isMetaClass = "Meta".equals(classDef.name().name());
+    if (!isMetaClass) {
```

Comment:
Minor nitpick but I believe you can inline this.

Reply:
inlined